homeassistant/components/binary_sensor/aurora.py

```diff
@@ -0,0 +1,155 @@
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the aurora sensor."""
+    if None in (hass.config.latitude, hass.config.longitude):
+        _LOGGER.error("Lat. or long. not set in Home Assistant config")
+        return False
+
+    name = config.get(CONF_NAME)
+    threshold = config.get(CONF_THRESHOLD)
+
+    try:
+        aurora_data = AuroraData(
+            hass.config.latitude,
+            hass.config.longitude,
+            threshold
+        )
+        aurora_data.update()
+    except requests.exceptions.HTTPError as error:
+        _LOGGER.error(
+            "Connection to aurora forecast service failed: %s", error)
+        return False
+
+    add_devices([AuroraSensor(aurora_data, name)], True)
+
+
+class AuroraSensor(BinarySensorDevice):
+    """Implementation of an aurora sensor."""
+
+    def __init__(self, aurora_data, name):
+        """Initialize the sensor."""
+        self.aurora_data = aurora_data
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{}'.format(self._name)
+
+    @property
+    def is_on(self):
+        """Return true if aurora is visible."""
+        return self.aurora_data.is_visible if self.aurora_data else False
+
+    @property
+    def sensor_class(self):
+        """Return the class of this sensor."""
+        return DEFAULT_SENSOR_CLASS
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attrs = {}
+
+        if self.aurora_data:
+            attrs["visibility_level"] = self.aurora_data.visibility_level
+            attrs["message"] = self.aurora_data.is_visible_text
+
+        return attrs
+
+    def update(self):
+        """Get the latest data from Aurora API and updates the states."""
+        self.aurora_data.update()
+
+
+class AuroraData(object):
+    """Get aurora forecast."""
+
+    def __init__(self, latitude, longitude, threshold):
+        """Initialize the data object."""
+        self.api_gateway = AuroraAPIGateway(latitude, longitude)
+        self.threshold = int(threshold)
+        self.is_visible = None
+        self.is_visible_text = None
+        self.visibility_level = None
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data from the Aurora service."""
+        try:
+            self.visibility_level = self.api_gateway.get_aurora_forecast()
+            if int(self.visibility_level) > self.threshold:
+                self.is_visible = True
+                self.is_visible_text = "visible!"
+            else:
+                self.is_visible = False
+                self.is_visible_text = "nothing's out"
+
+        except requests.exceptions.HTTPError as error:
+            _LOGGER.error(
+                "Connection to aurora forecast service failed: %s", error)
+            return False
+
+
+class AuroraAPIGateway(object):
```

Comment:
It is not allow to have device code inside hass. That need to be export to a own library. In your case it is a simple request and you can do that inside data.update but not in a object than can be Export to a own library.